doc update for only applying the patches when using cuda (#633)
rapidsai/ucx-py: docs/source/install.rst

    ../contrib/configure-devel --prefix=$CONDA_PREFIX --with-cuda=$CUDA_HOME --enable-mt CPPFLAGS="-I/$CUDA_HOME/include"
    make -j install

.. note::

>     If you're running on a machine without CUDA then you can skip applying the patches

Reviewer: ```suggestion
    If you're running on a machine without CUDA then you _must NOT_ apply the patches.
```
